front/lib/oauth.ts

```diff
@@ -0,0 +1,49 @@
+import type { OAuthProvider, WorkspaceType } from "@dust-tt/types";
+
+import type { OAuthUseCase } from "@app/lib/api/oauth";
+
+export async function auth({
+  owner,
+  provider,
+  useCase,
+  dustClientFacingUrl,
+}: {
+  owner: WorkspaceType;
+  provider: OAuthProvider;
+  useCase: OAuthUseCase;
+  dustClientFacingUrl: string;
+}): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const oauthPopup = window.open(
```

Comment:
this is a client side library

Reply:
I know but in the past we had people importing from here in the BE rather than from `api/lib/`. FWIW, we should never use window outside of a page or components IMHO.

Reply:
Well a client library that calls window is perfectly acceptable right? the only question is where to put it no?

Reply:
This client library will be called from pages or components

Reply:
Yes!